edb/server/mng_port/edgecon.pyx

```diff
@@ -1299,8 +1357,26 @@ cdef class EdgeConnection:
 
         out_buf.write_int16(<int16_t>argsnum)
 
-        in_len = frb_get_len(&in_buf)
-        out_buf.write_cstr(frb_read_all(&in_buf), in_len)
+        if array_tids:
+            # we have array parameters, ensure all of them
+            # have correct element OIDs as per Postgres' expectations.
+            array_tid_map = {a[0]: a[1] for a in array_tids}
```

Comment:
Is there a way to avoid dict here?

Reply:
The only alternative seems to be doing an O(N) search on the passed tuple `len(array_tids)` times.  Iterating over the tuple isn't very convenient either.

Reply:
Can I store dicts in query units?